### ISSUE TITLE
About the annotation files

### ISSUE THREAD

**Issue by KaKa-101** [Chat-3D/Chat-3D-v2#23]
Hi, could you tell me or release the code about how you generate each file in annotations (you have provided) during the data preprocessing. This confused me a lot because I didn't quite figure out what each of these files corresponded to. Thanks a lot for your help.

**Comment by ZzZZCHS** [comment]
We are refining our code recently and will re-organize these annotations and instructions later. (hopefully within two weeks) The main changes include: simplifying the training & evaluation process; support for vicuna v1.5 and LoRA; add special tokens to represent object ids. Stay tuned.

As for these annotation files, you can download them to check their content and follow the training guide to see their usage in different training stages. Basically, current annotations are all originated from scannet / scanrefer dataset. We will organize and provide the generation scripts later. If you have question about any specific file, you can raise a question here and I will explain it for you.

**Comment by KaKa-101** [comment]
Thanks for your quick reply. 
1. May I ask what is the difference between "pcd_by_instance" and "pcd_feats_ulip2" during the preprocess. It seems that the former is the point cloud of each object from all the scenes in ground truth of original dataset. So is the latter also point cloud of each object?
2. How do you segment & assign ID for each object in the pipeline? Do you just use the ground truth annotations of original datasets or do you preliminary train a 3D instance segmentation model pre-training on the GT annotations? And if I switch it to a new dataset, can it achieve good open-vocabulary segmentation performance?
Appreciate for your nice help.

**Comment by ZzZZCHS** [comment]
1. `pcd_by_instance` is the point cloud of each object. `pcd_feats_ulip2` is object features extracted by ulip2 for each object.
2. During training, we use two kinds of ID assignment. One is directly from GT labels. The other one is from PointGroup segmentation results. For inference, we only use PointGroup segmentation results. We choose the pretrained [model](https://github.com/dvlab-research/PointGroup?tab=readme-ov-file#pretrained-model) on scannet provided by PointGroup. I think you need to switch to a open-vocabulary segmentation model for evaluating on a new dataset.

**Comment by KaKa-101** [comment]
Thanks a lot. I also have the following questions:

1. How do you get the file "scannet_train_attributes.pt" and what's the function of this file? Why you don't need these two files like  "scanrefer_train_attributes.pt" or "nr3d_train_attributes.pt", which correspond to datasets Scanrefer and Nr3D.
2. What are the functions of each of these two files "scanrefer_train_stage1.json" and "scannet_train_stage1.json"? I noticed that for each object's captions, the former has multiple descriptions, while the latter only has one description, so why such difference exists? Are these two files just used to tell the model the class of the object?

Thanks again and appreciate for your help.
